src/core/layout/qgslayoutitemlegend.h

```diff
@@ -509,6 +587,8 @@ class CORE_EXPORT QgsLayoutItemLegend : public QgsLayoutItem
 
     friend class QgsCompositionConverter;
 
+    mutable QgsExpressionContext mExpContext = QgsLayoutItem::createExpressionContext();
```

Comment:
I'm not sure this member variable is required.

What I would propose:

* remove it
* remove the method updateExpressionContext()
* instead of calling `updateExpressionContext()` call `mLegendModel->setExpressionContext( createExpressionContext() )` directly

Reply:
Done

Reply:
@m-kuhn I went overboard and broke things in my code, should be fine now.